replica/replica.go

```diff
@@ -929,6 +929,9 @@ func (r *Replica) revertDisk(parent, created string) (*Replica, error) {
 }
 
 func (r *Replica) createDisk(name string, userCreated bool, created string) error {
+	if err := r.SyncDir(); err != nil {
```

Comment:
why we are calling this in the beginning of the function?

Reply:
this function is called to create snapshots. To ensure that the snapshot file has all the data and metadata synced to disk we are syncing the directory before creating snapshot. When snapshots are created IOs are paused, therefore at this point there are no IOs running in parallel.